download-maven-plugin/src/main/java/com/googlecode/WGet.java

```diff
@@ -215,12 +221,46 @@ public void execute() throws MojoExecutionException, MojoFailureException {
     // DO
     try
     {
-      if (outputFile.exists() && !overwrite)
+      boolean haveFile = outputFile.exists();
+      if (haveFile)
       {
+        if (this.checkSignature && this.md5 != null)
+        {
+          try {
+            SignatureUtils.verifySignature(outputFile, this.md5,
+                    MessageDigest.getInstance("MD5"));
+          } catch (MojoFailureException e) {
+            getLog().warn("File already exist. " + e.getMessage());
+            haveFile = false;
+          }
+        }
+        if (this.checkSignature && this.sha1 != null)
+        {
+          try {
+            SignatureUtils.verifySignature(outputFile, this.sha1,
+                    MessageDigest.getInstance("SHA1"));
+          } catch (MojoFailureException e) {
+            getLog().warn("File already exist. " + e.getMessage());
+            haveFile = false;
+          }
+        }
+        if (this.checkSignature && this.sha512 != null) {
+          try {
+            SignatureUtils.verifySignature(outputFile, this.sha512, MessageDigest.getInstance("SHA-512"));
+          } catch (MojoFailureException e) {
+            getLog().warn("File already exist. " + e.getMessage());
+            haveFile = false;
+          }
+        }
+
         // TODO verify last modification date
-        getLog().info("File already exist, skipping");
+        if (!haveFile)
```

Comment:
Please always use { ... } blocks even for single instruction after a control statement. It prevents for future bugs with people adding instructions and not thinking about adding { ... } as well (that happened to me more than once ;)